test/suite/test_timestamp08.py

```diff
@@ -0,0 +1,261 @@
+
+        # Take a checkpoint.  Make a copy of the database.  Open the
+        # copy and verify whether or not the expected data is in there.
+        self.pr("CKPT: " + self.ckptcfg)
+        ckptcfg = self.ckptcfg
+
+        self.session.checkpoint(ckptcfg)
+        self.backup_check(check_value, valcnt_ts_log, valcnt_ts_nolog)
+
+    def test_timestamp08(self):
+        if not wiredtiger.timestamp_build():
+            self.skipTest('requires a timestamp build')
+
+        config_default = 'key_format=i,value_format=i,memory_page_max=32k,leaf_page_max=8k,internal_page_max=8k'
+        config_nolog   = ',log=(enabled=false)'
+
+        #
+        # Open two timestamp tables:
+        # 1. Table is logged and uses timestamps.
+        # 2. Table is not logged and uses timestamps.
+        #
+        self.session.create(self.table_ts_log, 'key_format=i,value_format=i')
+        cur_ts_log = self.session.open_cursor(self.table_ts_log)
+        self.session.create(self.table_ts_nolog, 'key_format=i,value_format=i,log=(enabled=false)')
+        cur_ts_nolog = self.session.open_cursor(self.table_ts_nolog)
+
+        # Insert keys 1..100
+        nkeys = 100
+        orig_keys = range(1, nkeys+1)
+        keys = orig_keys[:]
+        random.shuffle(keys)
+
+        self.session.begin_transaction()
+        # Make three updates with different timestamps.
+        self.session.timestamp_transaction('commit_timestamp=' + timestamp_str(1))
+        for k in keys:
+            cur_ts_log[k] = 1
+            cur_ts_nolog[k] = 1
+
+        self.session.timestamp_transaction('commit_timestamp=' + timestamp_str(101))
+        for k in keys:
+            cur_ts_log[k] = 2
+            cur_ts_nolog[k] = 2
+
+        self.session.timestamp_transaction('commit_timestamp=' + timestamp_str(201))
+        for k in keys:
+            cur_ts_log[k] = 3
+            cur_ts_nolog[k] = 3
+
+        self.session.commit_transaction('commit_timestamp=' + timestamp_str(301))
+
+        # Scenario: 1
+        # Check that we see all the latest values (i.e. 3) as per transaction
+        # visibility when reading with out the read_timestamp.
+        # All tables should see all the values.
+        self.check(self.session, "", self.table_ts_log,
+            dict((k, 3) for k in orig_keys))
+        self.check(self.session, "", self.table_ts_nolog,
+            dict((k, 3) for k in orig_keys))
+
+        # Scenario: 2
+        # Set oldest and stable timestamps
+        self.oldts = timestamp_str(100)
+        # Set the stable_timestamp such that last update is beyond it.
+        self.stablets = timestamp_str(200)
+        self.conn.set_timestamp('oldest_timestamp=' + self.oldts)
+        self.conn.set_timestamp('stable_timestamp=' + self.stablets)
+
+        # Check that we see the values correctly till stable_timestamp.
+        self.check(self.session, 'read_timestamp=' + self.stablets,
+            self.table_ts_log,dict((k, 2) for k in orig_keys))
```

Comment:
space between "self.table_ts_log," and "dict((k, 2) for k in orig_keys". Same for the below line too

Reply:
updated